tuna/miopen/fin_class.py

```diff
@@ -622,6 +667,48 @@ def process_fdb_w_kernels(self,
 
     return status
 
+  def add_sql_objs(self, session, obj_list):
+    """add sql objects to the table"""
+    self.logger.info("adding pending #objects: %s", len(obj_list))
+    for sql_obj in obj_list:
+      session.add(sql_obj)
+    session.commit()
+    return True
+
+  def result_queue_commit(self, session, close_job):
+    """commit the result queue and set mark job complete"""
+    for_commit = {}
+    while not self.result_queue.empty():
+      job, sql_obj = self.result_queue.get(True, 1)
+      if job.id not in for_commit:
+        for_commit[job.id] = {"job": job, "obj": [sql_obj]}
+      else:
+        for_commit[job.id]['obj'].append(sql_obj)
+
+    this_job = self.job
+    for job_id, job_dict in for_commit.items():
+      obj_list = job_dict['obj']
+      self.job = job_dict['job']
+
+      self.logger.info("commit pending job %s, #objects: %s", job_id,
+                       len(obj_list))
+      status = session_retry(session, self.add_sql_objs,
+                             lambda x: x(session, obj_list), self.logger)
+      if not status:
+        self.logger.error("Failed commit pending job %s", job_id)
+        continue
+
+      #set job states after successful commit
+      close_job()
+
+    self.job = this_job
```

Comment:
why are we saving the current job in this_job. And why are we setting self.job=job_dict['job'] in the loop?